File: Kotlin/2.kt
fun countRounds(teams: Int): Int {
    if (teams == 1) {
        return 0
    }
    else if (teams % 2 == 0) {
        var matches = teams / 2
        return matches + countRounds(matches)
    }
    else {
        var matches = (teams-1) / 2
        return matches + countRounds(matches+1)
    }
}

fun main() {
    val totalTeams = readLine()!!.toInt()
    println(countRounds(totalTeams))
}

File: PHP/2.php
<?php
function countRounds($teams) {
    if ($teams == 1) {
        return 0;
    }
    elseif ($teams % 2 == 0) {
        $matches = $teams / 2;
        return $matches + countRounds($matches); 
    }
    else {
        $matches = ($teams - 1) / 2;
        return $matches + countRounds($matches+1);
    }
}

$total_teams = readline();
echo countRounds($total_teams);
?>